libos/src/fs/libos_fs_lock.c

```diff
@@ -154,10 +154,17 @@ static struct posix_lock* posix_lock_find_conflict(struct fs_lock* fs_lock, stru
     assert(pl->type != F_UNLCK);
 
     struct posix_lock* cur;
-    LISTP_FOR_EACH_ENTRY(cur, &fs_lock->posix_locks, list) {
-        if (cur->pid != pl->pid && pl->start <= cur->end && cur->start <= pl->end
-               && (cur->type == F_WRLCK || pl->type == F_WRLCK))
-            return cur;
+    if (pl->handle_id == 0) {
+        LISTP_FOR_EACH_ENTRY(cur, &fs_lock->posix_locks, list) {
```

Comment:
What happens to pending requests?

Reply:
Do you mean we should search the request list to find the conflicts?

Reply:
I meant pending POSIX lock requests `posix_lock_requests`.

Reply:
Sorry, Can you explain the problems with my code clearly?

Reply:
I would have assumed that you need to remove pending `posix_lock_requests` as seen in my PR from last week. Something like the following in the `else` statement. Could be wrong though. 

```
LISTP_FOR_EACH_ENTRY_SAFE(req, req_tmp, &fs_lock->posix_lock_requests, list) ...
```

Reply:
I would have assumed that we need to remove pending POSIX lock requests in the `else` condition as seen in my PR from last week. Something like the following. Might be wrong though.

```
LISTP_FOR_EACH_ENTRY_SAFE(req, req_tmp, &fs_lock->posix_lock_requests, list) ...
```

Reply:
We don't need to remove the pending request for they won't block the current lock request. If we didn't set `LOCK_NB` our `posix_lock` will be a request and added to the `fs_lock`'s request list. The requests will be processed (remove or become a `posix_lock` in `fs_lock`'s `posix_locks`) only when the `fs_lock`'s lock list changes.

In a child process, the `LOCK_NB` is also not set. The conflict `posix_lock` will send via IPC and let the leader process to let the `posix_lock` become a request.

Reply:
Thanks that clarifies a lot!